src/App.tsx

```diff
@@ -15,71 +15,92 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string,
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+// eslint-disable-next-line react/prefer-stateless-function
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  render() {
+    const { selectedGood } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        <h1
+          className={
+            selectedGood.length > 0
+              ? 'title is-flex is-align-items-center'
+              : 'title'
+          }
+        >
+          {
+            selectedGood.length > 0
+              ? `${selectedGood} is selected`
+              : 'No goods selected'
+          }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+          {selectedGood.length > 0
+            && (
+            /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => {
+                  this.setState({ selectedGood: '' });
+                }}
+              />
+            )}
+        </h1>
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map((good) => (
+              <tr
+                data-cy="Good"
+                className={
+                  selectedGood === good
+                    ? 'has-background-success-light'
+                    : ''
+                }
+              >
+                <td>
+                  <button
+                    data-cy={
+                      selectedGood === good
+                        ? 'RemoveButton'
+                        : 'AddButton'
+                    }
+                    type="button"
+                    className={
+                      selectedGood === good
+                        ? 'button is-info'
+                        : 'button'
```

Comment:
in both situations you need 'button' class, so you can use classNames, to avoid rewriting classes